### PR TITLE
Removed NetService.ProcessSegment by inlining the NetManager call.

### PR DESCRIPTION
**Pull request by DaEgi01**
Please double check that I always passed the correct segmentId to the NetManager and that the following segment reference is the one that is used further on.

### CODE REVIEW

**Review by kvakvs — COMMENTED** [review]
Looks good, as long as the changes are equivalent to the old code

**Comment by kianzarrin** [comment]
@DaEgi01 wouldn't it have been better to just use `segmentID.ToSegment()` from the extension class called `Shortcuts` ?
is this code in the performance critical part of the code?

**Comment by DaEgi01** [comment]
> Looks good, as long as the changes are equivalent to the old code

It should be. I checked three times but I m not perfect.
So it would be great if another pair of eyes could verify it.

> @DaEgi01 wouldn't it have been better to just use `segmentID.ToSegment()` from the extension class called `Shortcuts` ?
> is this code in the performance critical part of the code?

Probably, I forgot about ToSegment.
The original method that I replaced did not use the ToSegment extension method.
I tried to just do the conversion mechanically without any extras such that I don't introduce any errors.
But I did a cleanup here and there - for example I inlined the previously separate declaration and initialization of some variables, which was necessary before but not after - less assignments is also better :)
Regarding perf, my first assumption would be that the ToSegment() method is probably simple enough that it will be inlined during compilation. I should double check that.
If not, we can still try with the aggresive inlining attribute.
I ll check that.

**Comment by DaEgi01** [comment]
Ok, moved the code to "ToSegment()".
I checked if it is inlined after compilation, and it is not.
Also `[MethodImpl(MethodImplOptions.AggressiveInlining)]` seems to be a .net 4.5+ thing, so we are out of luck here.
I would still keep using ToSegment() since its handy and the perf cost of a static method call is not high.
Its significantly better than the closure before, that is for sure.
Also Singleton<NetManager>.instance.m_segments.m_buffer is already dereferenced in _segBuffer so its probably even a little faster (maybe that is the reason why it can't be inlined!?).

**Review by FireController1847 — APPROVED** [review]
LGTM 👍

File: TLM/TLM/Manager/Impl/ExtVehicleManager.cs [hunk 1]
```diff
@@ -837,23 +838,20 @@ public class ExtVehicleManager
...
 
-            ushort transitNodeId2 = 0;
-            Constants.ServiceFactory.NetService.ProcessSegment(
-                nextPos.m_segment,
-                (ushort segmentId, ref NetSegment segment) => {
-                    transitNodeId2 = startNode ? segment.m_startNode : segment.m_endNode;
-                    return true;
-                });
+            ref NetSegment curPosSegment = ref curPos.m_segment.ToSegment();
+            var transitNodeId1 = startNode
```
Comment (krzychu124) [inline TLM/TLM/Manager/Impl/ExtVehicleManager.cs:843]: Hmm, I believe there is a method in `NetSegment`  returning start/end node 🤔

**Review by krzychu124 — CHANGES_REQUESTED** [review]

File: TLM/TLM/Manager/Impl/ExtVehicleManager.cs (hunk 1 above)
Comment (krzychu124) [reply TLM/TLM/Manager/Impl/ExtVehicleManager.cs:843]: Ok, nvm, I couldn't find it... 😅

**Review by krzychu124 — APPROVED** [review]
Since I couldn't find method from requested I added, I approve PR. If you want to implement it as extension method, go ahead :)

**Pull request merged**